conans/test/integration/command_v2/test_outdated.py

```diff
@@ -0,0 +1,228 @@
+import json
+from collections import OrderedDict
+
+from conans.test.assets.genconanfile import GenConanfile
+from conans.test.utils.tools import TestClient, TestServer
+
+
+def test_outdated_command():
+    tc = TestClient(default_server_user=True)
+    # Create libraries needed to generate the dependency graph
+    tc.save({"conanfile.py": GenConanfile()})
+    tc.run("create . --name=zlib --version=1.0")
+
+    tc.run("create . --name=foo --version=1.0")
+    tc.save({"conanfile.py": GenConanfile().with_requires("foo/[>=1.0]")})
+    tc.run("create . --name=libcurl --version=1.0")
+
+    # Upload the created libraries to remote
+    tc.run("upload * -c -r=default")
+
+    # Create new version of libraries in remote and remove them from cache
+    tc.save({"conanfile.py": GenConanfile()})
+    tc.run("create . --name=foo --version=2.0")
+    tc.run("create . --name=zlib --version=2.0")
+    tc.run("upload * -c -r=default")
+    tc.run("remove foo/2.0 -c")
+    tc.run("remove zlib/2.0 -c")
+
+    tc.save({"conanfile.py": GenConanfile("app", "1.0").with_requires("zlib/1.0", "libcurl/[>=1.0]")})
+    # tc.run("graph info . --update")
+    tc.run("graph outdated . --format=json")
+    output = json.loads(tc.stdout)
+    assert "zlib" in output
+    assert "foo" in output
+    assert "libcurl" not in output
+    assert output["zlib"]["current_versions"] == ["zlib/1.0"]
+    assert output["zlib"]["version_ranges"] == []
+    assert output["zlib"]["latest_remote"]["ref"] == "zlib/2.0"
+    assert output["zlib"]["latest_remote"]["remote"].startswith("default")
+    assert output["foo"]["current_versions"] == ["foo/1.0"]
+    assert output["foo"]["version_ranges"] == ["foo/[>=1.0]"]
+    assert output["foo"]["latest_remote"]["ref"] == "foo/2.0"
+    assert output["foo"]["latest_remote"]["remote"].startswith("default")
+
+
+def test_recipe_with_lockfile():
+    tc = TestClient(default_server_user=True)
+    # Create libraries needed to generate the dependency graph
+    tc.save({"conanfile.py": GenConanfile()})
+    tc.run("create . --name=zlib --version=1.0")
+    tc.run("create . --name=foo --version=1.0")
+
+    tc.save({"conanfile.py": GenConanfile().with_requires("foo/[>=1.0]")})
+    tc.run("create . --name=libcurl --version=1.0")
+
+    # Upload the created libraries to remote
+    tc.run("upload * -c -r=default")
+
+    # Create new version of libraries in remote and remove them from cache
+    tc.save({"conanfile.py": GenConanfile()})
+    tc.run("create . --name=foo --version=2.0")
+    tc.run("create . --name=zlib --version=2.0")
+    tc.run("upload * -c -r=default")
+    tc.run("remove foo/2.0 -c")
+    tc.run("remove zlib/2.0 -c")
+
+    tc.save({"conanfile.py": GenConanfile("app", "1.0").with_requires("zlib/1.0",
```

Comment:
Repeated to previous test, consider refactoring. Most likely you want a test that is parameterized, if is is very similar, just with or without lockfile.